okhttp/src/main/java/okhttp3/MediaType.java

```diff
@@ -45,20 +45,29 @@ private MediaType(String mediaType, String type, String subtype, @Nullable Strin
   }
 
   /**
-   * Returns a media type for {@code string}, or null if {@code string} is not a well-formed media
-   * type.
+   * Returns a media type for {@code string}.
+   *
+   * @throws IllegalArgumentException if {@code string} is not a well-formed media type.
    */
-  public static @Nullable MediaType parse(String string) {
+  public static MediaType parseOrThrow(String string) {
```

Comment:
I would like this method to be named `get()`:

 * parse(String s): Attempt to get s as a media type, returning null if it isn't one.
 * get(String s): Returns the media type for s, throwing if it isn’t one.

And we can do likewise for HttpUrl.

Reply:
Oh, yeah. Nice.

For whatever reason I thought `HttpUrl` used `get` already.

Reply:
It does for URL and URI. They could have a different scheme (say ftp:) so these methods return null. For OkHttp 4.x I think we change these methods to throw instead of returning null.